src/main/java/net/dv8tion/jda/api/utils/messages/MessageCreateBuilder.java

```diff
@@ -0,0 +1,288 @@
+import java.util.List;
+
+/**
+ * Builder specialized for building a {@link MessageCreateData}.
+ * <br>This can be used to build a request and send it to various API endpoints.
+ *
+ * <p><b>Example</b>
+ * <pre>{@code
+ * try (FileUpload file = FileUpload.fromData(new File("wave.gif"))) {
+ *     MessageCreateData data = new MessageCreateBuilder()
+ *       .setContent("Hello guys!")
+ *       .setTTS(true)
+ *       .setFiles(file)
+ *       .build();
+ *
+ *     for (MessageChannel channel : channels) {
+ *         channel.sendMessage(data).queue();
+ *     }
+ * } // closes wave.gif if an error occurred
+ * }</pre>
+ *
+ * @see net.dv8tion.jda.api.entities.MessageChannel#sendMessage(MessageCreateData) MessageChannel.sendMessage(data)
+ * @see net.dv8tion.jda.api.interactions.callbacks.IReplyCallback#reply(MessageCreateData) IReplyCallback.reply(data)
+ * @see net.dv8tion.jda.api.interactions.InteractionHook#sendMessage(MessageCreateData) InteractionHook.sendMessage(data)
+ * @see MessageEditBuilder
+ */
+public class MessageCreateBuilder extends AbstractMessageBuilder<MessageCreateData, MessageCreateBuilder> implements MessageCreateRequest<MessageCreateBuilder>
+{
+    private final List<FileUpload> files = new ArrayList<>(10);
+    private boolean tts;
+    private int flags = 0;
+
+    public MessageCreateBuilder() {}
+
+    /**
+     * Factory method to start a builder from an existing instance of {@link MessageCreateData}.
+     * <br>Equivalent to {@code new MessageCreateBuilder().applyData(data)}.
+     *
+     * @param  data
+     *         The message create data to apply
+     *
+     * @throws IllegalArgumentException
+     *         If null is provided
+     *
+     * @return A new MessageCreateBuilder instance with the applied data
+     *
+     * @see    #applyData(MessageCreateData)
+     */
+    @Nonnull
+    public static MessageCreateBuilder from(@Nonnull MessageCreateData data)
+    {
+        return new MessageCreateBuilder().applyData(data);
+    }
+
+    /**
+     * Factory method to start a builder from an existing instance of {@link MessageEditData}.
+     * <br>Equivalent to {@code new MessageCreateBuilder().applyEditData(data)}.
+     * <br>This will only set fields which were explicitly set on the {@link MessageEditBuilder},
+     * unless it was configured to be {@link MessageEditRequest#replace(boolean) replacing}.
+     *
+     * <p>This will <b>not</b> copy the message's attachments, only any configured {@link FileUpload FileUploads}.
+     * To copy attachments, you must download them explicitly instead.
+     *
+     * @param  data
+     *         The message edit data to apply
+     *
+     * @throws IllegalArgumentException
+     *         If null is provided
+     *
+     * @return A new MessageCreateBuilder instance with the applied data
+     *
+     * @see    #applyEditData(MessageEditData)
+     */
+    @Nonnull
+    public static MessageCreateBuilder fromEdit(@Nonnull MessageEditData data)
```

Comment:
Question: Why are we doing `from` + `fromEdit` + `fromMessage` instead of just `from`?
We could just overload.

Just for developer discovery?

Reply:
Its consistent with other factory methods like on Emoji